crates/ty_python_semantic/src/site_packages.rs

```diff
@@ -240,6 +247,52 @@ impl PythonEnvironment {
     }
 }
 
+/// Enumeration of the subdirectories of `sys.prefix` that could contain a
+/// `site-packages` directory if the host system is Unix-like.
+///
+/// For example, if `sys.prefix` is `.venv` and the Python version is 3.10,
+/// the `site-packages` directory could be located at `.venv/lib/python3.10/site-packages`,
+/// or at `.venv/lib64/python3.10/site-packages`, or there could indeed be `site-packages`
+/// directories at both of these locations.
+#[derive(Debug, Clone, Copy, Eq, PartialEq, strum_macros::EnumIter)]
+enum UnixLibDir {
+    Lib,
+    Lib64,
+}
+
+impl UnixLibDir {
+    const fn as_str(self) -> &'static str {
+        match self {
+            Self::Lib => "lib",
+            Self::Lib64 => "lib64",
+        }
+    }
+}
+
+impl std::fmt::Display for UnixLibDir {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(self.as_str())
+    }
+}
+
+impl AsRef<SystemPath> for UnixLibDir {
+    fn as_ref(&self) -> &SystemPath {
+        SystemPath::new(self.as_str())
+    }
+}
+
+impl PartialEq<&str> for UnixLibDir {
+    fn eq(&self, other: &&str) -> bool {
+        self.as_str() == *other
+    }
+}
```

Comment:
Does this work
```suggestion
impl PartialEq<str> for UnixLibDir {
    fn eq(&self, other: &str) -> bool {
        self.as_str() == other
    }
}
```

Reply:
if I do that then it makes the places where I'm using the `impl` less ergonomic; I have to do awkward dereferences